packages/calcite-components/src/utils/number.ts

```diff
@@ -217,3 +226,30 @@ export function expandExponentialNumberString(numberString: string): string {
 function stringContainsNumbers(string: string): boolean {
   return numberKeys.some((number) => string.includes(number));
 }
+
+/**
+ * Adds localized trailing decimals zero values to the number string.
+ * BigInt conversion to string removes the trailing decimal zero values (Ex: 1.000 is returned as 1). This method helps adding them back.
+ *
+ * @param {string} localizedValue - localized number string value
+ * @param {string} value - current value in the input field
+ * @param {NumberStringFormat} numberStringFormatter - numberStringFormatter instance to localize the number value
+ * @returns {string} localized number string value
+ */
+export function addLocalizedTrailingDecimalZeros(
+  localizedValue: string,
+  value: string,
+  formatter: NumberStringFormat
+): string {
+  const decimalSeparator = formatter.decimal;
+  const localizedIntegers = localizedValue.includes(decimalSeparator)
+    ? localizedValue.split(decimalSeparator)[0]
+    : localizedValue;
+  const decimals = value.split(".")[1];
+
+  return decimals
+    ? `${localizedIntegers}${decimalSeparator}${Array.from(decimals)
+        .map((d) => formatter.localize(d))
```

Comment:
I'm still not sure this is what we want, could it potentially add non 0 decimal places? Could you add some spec tests including numbers like `123.456000`. If we are iterating over all of the decimal places it would end up appending the `456` as well right? When we only want to add back the `000`.

Reply:
Yeah, i thought about it. When i was trying to add just the missing zeros the function is getting a bit more complex but i see your point where this function localizes characters which are localized by then resulting in redundancy. It can be an issue for super long decimals. 

how about this 

```onst decimalSeparator = formatter.decimal;
  const decimals = value.split(".")[1];
  const localizedDecimals = localizedValue.split(decimalSeparator)[1];

  const diff = decimals && localizedDecimals ? decimals.length - localizedDecimals.length : 0;
  return localizedValue.padEnd(localizedValue.length + diff, formatter.localize("0"));
}

Reply:
never mind, the above will fail

Reply:
What you probably want is a regex expression like `/\..*0$/`. to check for trailing zeros after a decimal, but excluding non-zero numbers after the decimal

Reply:
Updating/simplified the regex. We don't care about the actual decimal because it is already being split, so we just need to grab the trailing zeros and append them to the localized value